src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Routing/SSRRedirectionStreaming.razor

```diff
@@ -0,0 +1,19 @@
+﻿@page "/routing/ssr-streaming-navigate-to"
+@attribute [StreamRendering]
+@using Microsoft.AspNetCore.Components.Forms
+@inject NavigationManager NavigationManager
+
+<p id="test-info">Click submit to navigate to home</p>
+<form method="post" @onsubmit="Submit" @formname="MyUniqueFormName">
+    <AntiforgeryToken />
+    <button type="submit" id="redirectButton" class="btn btn-primary">Redirect</button>
+</form>
+
+@code {
+    private async Task Submit()
+    {
+        await Task.Delay(1000); // Simulate some async work to let the streaming begin
```

Comment:
```suggestion
        await Task.Yield();
```

Whenever we introduce delays like this we slow down the entire test suite and these delays compound over our big collections. There are two ways to handle this:
* Use `Task.Yield()` if you are not checking any intermediate content.
* Otherwise, use a TaskCompletionSource and use a separate Http request to the server to complete it (using http client from the test, there are some tests that do this)